### PR TITLE
Remove always-false comparison in Spark version checker

### PR DESCRIPTION
**Pull request by fbiville**
Caught via a compiler warning

### CODE REVIEW

File: common/src/main/scala/org/neo4j/spark/util/Validations.scala [hunk 1]
```diff
@@ -53,7 +53,7 @@ case class ValidateSparkVersion(supportedVersions: String*) extends Validation {
     val splittedVersion = sparkVersion.split("\\.")
 
     ValidationUtil.isTrue(
-      sparkVersion(0) == "UNKNOWN" || supportedVersions
```
Comment (fbiville) [inline common/src/main/scala/org/neo4j/spark/util/Validations.scala:56]: this was comparing a char with a string

File: common/src/main/scala/org/neo4j/spark/util/Validations.scala [hunk 2]
```diff
@@ -53,7 +53,7 @@ case class ValidateSparkVersion(supportedVersions: String*) extends Validation {
...
+      supportedVersions
```
Comment (conker84) [inline common/src/main/scala/org/neo4j/spark/util/Validations.scala]: I think that maybe we can improve it in this way.
```suggestion
      sparkVersion == "UNKNOWN" || supportedVersions
```

If for some reason we cannot retrieve the version we don't want to block the user.

**Review by conker84 — COMMENTED** [review]
Thank you so much @fbiville please check my comment

**Review by conker84 — APPROVED** [review]

**Comment by conker84** [comment]
@fbiville thank you so much!

**Pull request merged**